### PR TITLE
Refactor radare2 commands usage in Cutter using cmdRaw and cmdRawAt

### PR DESCRIPTION
**Pull request by ITAYC0HEN**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://cutter.re/docs/developers-docs/first-time.html) to this repository
- [x] I made sure to follow the project's [coding style](https://cutter.re/docs/developers-docs.html)
- [x] I've updated the [documentation](https://cutter.re/docs/user-docs.html) with the relevant information (if needed)

**Detailed description**

This pull-request fixes the current implementation of `cmdRaw` and re-implements it properly. A POC for this was already merged in `cmdRawAt`. All the possible occurrences of `cmd()` usage were replaced with `cmdRaw()` and `cmdRawAt()` commands to ensure safety and simplicity. This also improves performance since Cutter would not execute command-parsing in radare2. This solution should solve the command injections problem, but this needed to check thoroughly.

To make regression resolution easier, changes to each file were created separately. I'd be happy if they'll merge as-is and won't be changed.

Documentation was expanded and improved.

Sometimes I improved existing code to use QString properly, but sometimes not.

**Test plan (required)**

The pull request made major changes across the entire code-base and thus need to be tested by you, the reviewers, very thoroughly. Please pay attention especially to changes in dialogs and features that execute user-input in the commands, such as renaming or setting new flags.
Especially, make sure to test special and risky chars such as: ` @ ! " > ; - <
I usually checking for command injection such as "! ping 8.8.8.8" because then Cutter is freezing and it is much visible :) 

**AnalTask.cpp**
- [x] Changing asm.os in the open dialog
- [x] Open a shellcode in Cutter
  
**ColorThemeWorker.cpp**
- [x] Duplicate existing theme
- [x] Check that selected theme is loaded when opening Cutter

**Configuration.cpp**
- [x] Reset settings
- [x] Check that setting new color theme works

**MainWindow.cpp**
- [x] Check that opening a project works

**CommentsDialog.cpp**
- [x] Check that adding comment work
- [x] Check that editing an existing comment work
- [x] Test for special value in the comments (e.g @ " ' ` ; )
  

**EditVariableDialog.cpp**
*In this one, a bug was fixed `afn` -> `afn.`*

- [x] Check that the title of the function variable editing dialog shows correctly

**NewFileDialog.cpp**

- [x] Check that recent projects list shows recent items correctly

**WriteCommandsDialog.cpp**

- [x] Check that the Duplicate from offset dialog shows preview correctly

**ColorThemeEditor.cpp**

- [x] Make sure that changing a color of a config works in editing mode

**DisassemblyContextMenu.cpp**
- [x] Check if Remove string is shown instead of "add" when string is defined
- [x] Check that Edit Comment is shown instead Add Comment
- [X] Check that exisiting comment is set in the comment box when editing
- [x] Check if renaming "used here" works properly. Test for special input
- [x] Check if a list of calling conventions is shown during function editing
- [x] Check that editing calling convention applied

**CutterSamplePlugin.cpp**
- [x] Check if cmdRaw works on button click

**ColorThemListView.cpp**
- [x] Check if choosing a value to edit causes blinking in preview
- [x] Make sure that changing color applies

**Dashboard.cpp**
- [x] Make sure that entropy is calculated and showed correctly

**DebugActions.cpp**
- [x] Check that Continue until Main works properly

**DisassemblerGraphView.cpp**
- [x] Check that exporting r2 graph to graphviz works
- [x] Check that exporting r2 graph to dot file works
- [x] Check hat graph name shows name and prototype correctly

**HexdumpWidget.cpp**
- [x] Check that hashes for selected bytes showed correctly in the sidebar

**TypesWidget.cpp**
- [x] Check that exporting types works correctly

**Cutter.cpp**

TODO

**Closing issues**

### CODE REVIEW

File: src/core/Cutter.cpp [hunk 1]
```diff
@@ -699,72 +697,66 @@ void CutterCore::setAsString(RVA addr, int size, StringTypeFormats type)
...
     seekAndShow(addr);
-    QString arg;
-    if(size > 0) {
-        arg = QString::asprintf("%d @ %lld", size, addr);
-    } else {
-        arg = QString::asprintf("@ %lld", addr);
-    }
 
-    cmd(command + arg);
+    cmdRawAt(QString("%1 %2").arg(command).arg(size), addr);
```
[the remark this answers is not in the document]
Comment (karliss) [reply src/core/Cutter.cpp]: Looks like it works.

File: src/widgets/DebugActions.cpp [hunk 2]
```diff
@@ -191,6 +191,7 @@ DebugActions::DebugActions(QToolBar *toolBar, MainWindow *main) :
 
 void DebugActions::setButtonVisibleIfMainExists()
 {
+    // Use cmd because cmdRaw would not handle multiple commands concatenated
     int mainExists = Core()->cmd("f?sym.main; ??").toInt();
```
[the remark this answers is not in the document]
Comment (ITAYC0HEN) [reply src/widgets/DebugActions.cpp]: I actually tried that, but the return code of the _call function isn't reliable across different commands I tried.

[the remark this answers is not in the document]
Comment (karliss) [reply src/widgets/DebugActions.cpp]: Sure at the moment it's not very useful. That's why I said further iterations not right now. There were some discussions in r2 issue tracker about making error handling and the error codes more consistent.

File: src/core/Cutter.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (ITAYC0HEN) [reply src/core/Cutter.cpp]: yeah I checked :)

File: src/widgets/Dashboard.cpp [hunk 3]
```diff
@@ -105,7 +105,7 @@ void Dashboard::updateContents()
 
         // Calculate the Entropy of the entire binary from offset 0 to $s
         // where $s is the size of the entire file
-        QString entropy = Core()->cmd("ph entropy $s @ 0").trimmed();
+        QString entropy = Core()->cmdRawAt("ph entropy $s", 0).trimmed();
```
[the remark this answers is not in the document]
Comment (ITAYC0HEN) [reply src/widgets/Dashboard.cpp]: I don't know really

**Comment by karliss** [comment]
I tried some of the missing tests.

* ColorThemeWorker.cpp
	* Duplicate existing theme
	* Check that selected theme is loaded when opening Cutter
* Configuration.cpp
	* Reset settings
	* Check that setting new color theme works
* MainWindow.cpp
	 * Check that opening a project works -> correct executable opened, comment was there
* CommentsDialog.cpp
	* Check that adding comment wo
	* Check that editing an existing comment work
	* Test for special value in the comments (e.g @ " ' ` ; )
	 
* DisassemblerGraphView.cpp
	* Check that exporting r2 graph to dot file works
	* Check that exporting r2 graph to graphviz works -> How does this differ from dot? Exporting to png using grpahviz works.
   
* Check if renaming "used here" works properly. Test for special input -> something goes wrong when using symbols
* Check that Continue until Main works properly -> The executables I had didn't have sym.main only main. I don't think this is related to refactoring.

**Comment by karliss** [comment]
doing cmdRawAt("afn name", addr) currently doesn't work as good as expected because r2 internally builds and calls new command using `r_core_cmd0`.

**Comment by ITAYC0HEN** [comment]
is this something that can be fixed from Cutter? when you say "not correctly", you mean command injections? or doesn't work at all?

is it an easy fix in r2?

**Comment by karliss** [comment]
Yes, since it builds a r2 shell command command injection is possible. Fixing in r2 will take some time.

**Comment by karliss** [comment]
Fix for "rename used here" https://github.com/radareorg/radare2/pull/16280 . Will look into r2 tests tomorrow.

In cutter seemed to work even with special symbols. In case of flag they got replaced with _, but realname wasn't set. From the comment I expected realname to contain those symbols. When renaming function all the symbols were successfully set.

**Pull request merged**